flux-tests/tests/neg/surface/closure00.rs

```diff
@@ -0,0 +1,12 @@
+#![feature(register_tool)]
+#![register_tool(flux)]
+
+#[flux::sig(fn(c: Option<bool>) -> Option<i32{v:10 <= v}>)]
+pub fn test0(c: Option<bool>) -> Option<i32> {
+    c.map(|b| if b { 1 } else { 2 }) //~ ERROR: postcondition
+}
+
+#[flux::sig(fn(c: Option<bool>) -> Option<i32{v:1 <= v}>)]
+pub fn test1(c: Option<bool>) -> Option<i32> {
+    c.map(|b| if b { 1 } else { 2 })
+}
```

Comment:
This should go in `pos/surface`.

Maybe we can also add a slightly more interesting test like

```
#[flux::sig(fn(c: Option<bool[true]>) -> Option<i32[1]>)]
pub fn test1(c: Option<bool>) -> Option<i32> {
    c.map(|b| if b { 1 } else { 2 })
}
```

Reply:
done!